domain/model/service/modelservice.go

```diff
@@ -469,3 +474,10 @@ func EnvironVersionProviderGetter() EnvironVersionProviderFunc {
 		return environVersionProvider, nil
 	}
 }
+
+// IsControllerModel returns true if the model identified by the given UUID is the controller model.
+// The following errors may be returned:
+// - [modelerrors.NotFound] when no model by the given modelUUID exists.
+func (s *ModelService) IsControllerModel(ctx context.Context, uuid coremodel.UUID) (bool, error) {
```

Comment:
You don't need this change. Just call `GetModelInfo` and it is on the structure that is returned to you. You can then just get rid of the uuid on the api.